### DISCUSSION TITLE
trying to use Xsens DOT as IMU

### DISCUSSION THREAD

**Discussion by Jodutti** [PlusToolkit/PlusLib#1143] [Q&A] [answered]
Hello everybody,
I'm trying to use a Xsens DOT sensor to get the position of an ultrasound device in 3D Slicer. The Ultrasound is working but i don't get the IMU to work. This is currently my config file: . 
[PlusDevice_Server_XsensTracker.txt](https://github.com/PlusToolkit/PlusLib/files/13647920/PlusDevice_Server_XsensTracker.txt)
The xml file can be selected in the Device set selection but I get an error: Server stopped unexpectedly. Return code 1 in E:\D\PSNPTeb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(864)
I can't find this file to look up the error.
Do I need to do something else or is the config file enough if it's working?
I'm completly new in this topic and thankfull for every help! Thank you!
Kind regards

**Comment by Sunderlandkyl** [comment, marked as answer]
"Movella DOT" is not currently a supported device type.

**Reply by Jodutti** [reply]
Thank you very much!
In generell, is a config file enough to use an IMU in PlusServer?
Where can i check, if the IMU is supported?

**Reply by Sunderlandkyl** [reply]
Different devices may require different installers, but if your device is included then it is enough to modify the config file.
You can check which devices are included by checking the matrix on this page: https://plustoolkit.github.io/download (which is not exhaustive), or by opening plus and checking the log file (look for "Supported devices").

You can find the full list of supported devices here http://perk-software.cs.queensu.ca/plus/doc/nightly/user/Configuration.html
The "Inertial measurement units" should contain a list of currently supported IMUs.

We welcome any contributions if you have the time/resources to add new device support to Plus.

**Reply by Jodutti** [reply]
Thank you very much!

**Comment by Jodutti** [comment]
Recently, I came back to this topic and I would like to add the device.
Is it correct, that I need to wright a driver for this device? Or what do I have to do, to reach the goal?
Thanks for your time and your answer!

**Reply by Sunderlandkyl** [reply]
You would need to add a new "device" to Plus that interfaces with the the IMU.
This document has a description of the steps required: https://plustoolkit.github.io/devicecode.

It might also be useful to look at the code for other such devices: https://github.com/PlusToolkit/PlusLib/tree/master/src/PlusDataCollection

This folder contains the device for the Phidget 3/3/3 tracker:
https://github.com/PlusToolkit/PlusLib/tree/master/src/PlusDataCollection/PhidgetSpatialTracking
